packages/atomic/src/components/result-lists/result-list-common.tsx

```diff
@@ -340,8 +339,9 @@ export class ResultListCommon {
 }
 
 export type ResultRenderingFunction = (
-  result: Result | FoldedResult
-) => HTMLElement;
+  result: Result | FoldedResult,
+  root: HTMLElement
+) => string;
```

Comment:
I find the syntax with both the `root` parameter and the `string` return type a bit confusing. Would it be possible to use a syntax like `(result: Result | FoldedResult) => string | (result: Result | FoldedResult, root: HTMLElement) => void`, and infer the html content from the root's `innerHTML`? The return function could even optionally return a `Promise` that we await before reading the root's content.

Reply:
When `createRoot(root).render(template(rest));` is executed, `root.innerHTML` won't necessarily be rendered, it's really up to React internal, which decides when it schedule the paint.

I can check if there is a way to "be sure" that it's rendered.